docs/src/extend/custom-rule-tutorial.md

```diff
@@ -0,0 +1,461 @@
+                             * a message placeholder to include the incorrect value
+                             * in the error message.
+                             * Also includes a `fix(fixer)` function that replaces
+                             * any values assigned to `const foo` with "bar".
+                             */
+                            context.report({
+                                node,
+                                message: 'Value other than "bar" assigned to `const foo`. Unexpected value: {{ notBar }}.',
+                                data: {
+                                    notBar: node.init.value
+                                },
+                                fix(fixer) {
+                                    return fixer.replaceText(node.init, '"bar"');
+                                }
+                            });
+                        }
+                    }
+                }
+            }
+        };
+    }
+};
+{% endraw %}
+```
+
+## Step 5: Set up Testing
+
+With the rule written, you can test it to make sure it's working as expected.
+
+ESLint provides the built-in [`RuleTester`](../integrate/nodejs-api#ruletester) class to test rules. You do not need to use third-party testing libraries to test ESLint rules, but `RuleTester` works seamlessly with tools like Mocha and Jest.
+
+Next, create the file for the tests, `enforce-foo-bar.test.js`:
+
+```shell
+touch enforce-foo-bar.test.js
+```
+
+And add a test script to your `package.json` file to run the tests:
+
+```javascript
+// package.json
+{
+    // ...other configuration
+    "scripts": {
+        "test": "node enforce-foo-bar.test.js"
+    },
+    // ...other configuration
+}
+```
+
+## Step 6: Write the Test
+
+To write the test using `RuleTester`, import the class and your custom rule into the `enforce-foo-bar.test.js` file.
+
+The `RuleTester#run()` method tests the rule against valid and invalid test cases. If the rule fails to pass any of the test scenarios, this method throws an error.
+`RuleTester` requires that at least one valid and one invalid test scenarios be present.
+
+```javascript
+// enforce-foo-bar.test.js
+const {RuleTester} = require("eslint");
+const fooBarRule = require("./enforce-foo-bar");
+
+const ruleTester = new RuleTester({
+  // Must use at least ecmaVersion 2015 because
+  // that's when `const` variables were introduced.
+  parserOptions: { ecmaVersion: 2015 }
+});
+
+// Throws error if the tests in ruleTester.run() do not pass
+ruleTester.run(
+  "enforce-foo-bar", // rule name
+  fooBarRule, // rule code
+  { // checks
+    // 'valid' checks cases that should pass
+    valid: [{
+      code: "const foo = 'bar';",
+    }],
+    // 'invalid' checks cases that should not pass
+    invalid: [{
+      code: "const foo = 'baz';",
+      output: 'const foo = "bar";',
+      errors: 1,
+    }],
+  }
+);
+
+console.log("All tests passed!");
+```
+
```

Comment:
We described how to add a test file and a test script. It would be good to also describe how to run the tests (`npm test`).

Also, for this to work there should be a step where `eslint` is locally installed (`npm install --save-dev eslint`)

Reply:
````suggestion

Run the test with the following command:

```shell
npm test
```

You should see the following in your console:

```shell
All tests passed!
```

````